internal/server/add_assurance_visit_test.go

```diff
@@ -51,5 +79,116 @@ func TestPostAssuranceVisit(t *testing.T) {
 	testHandler.ServeHTTP(w, r)
 	resp := w.Result()
 	assert.Equal(http.StatusOK, resp.StatusCode)
+	assert.Equal(Redirect("/123/assurance-visits?success=addAssuranceVisit"), returnedError)
+}
+
+func TestAssuranceVisitHandlesValidationErrorsGeneratedWithinFile(t *testing.T) {
+	assert := assert.New(t)
+	client := &mockAddAssuranceVisitInformation{}
+
+	form := url.Values{}
+	form.Add("assurance-type", "")
+	form.Add("requested-date", "")
+
+	template := &mockTemplates{}
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/123/assurance-visits", strings.NewReader(form.Encode()))
+
+	var returnedError error
+
+	testHandler := mux.NewRouter()
+	testHandler.HandleFunc("/{id}/assurance-visits", func(w http.ResponseWriter, r *http.Request) {
+		returnedError = renderTemplateForAddAssuranceVisit(client, template)(sirius.DeputyDetails{}, w, r)
```

Comment:
I feel the `testHandler` is an unnecessary complication here (and in most other places I've seen it)